_drafts/practical_zephyr_05_west.md

```diff
@@ -0,0 +1,1246 @@
+
+Having all tools installed, the build should now pass, or shouldn't it? Worth a try!
+
+```bash
+$ source app/setup-sdk-nrf.sh
+$ west build --board nrf52840dk_nrf52840 app
+-- west build: generating a build system
+Loading Zephyr default modules (Zephyr base (cached)).
+-- Application: /path/to/workspace/app
+...
+-- Found host-tools: zephyr 0.16.0 (/opt/nordic/ncs/toolchains/4ef6631da0/opt/zephyr-sdk)
+-- Found toolchain: zephyr 0.16.0 (/opt/nordic/ncs/toolchains/4ef6631da0/opt/zephyr-sdk)
+...
+-- Including generated dts.cmake file: /path/to/workspace/build/zephyr/dts.cmake
+
+warning: USE_SEGGER_RTT (defined at modules/segger/Kconfig:12) was assigned the value 'y' but got
+the value 'n'. Check these unsatisfied dependencies: HAS_SEGGER_RTT (=n), 0 (=n). See
+http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_USE_SEGGER_RTT and/or look up
+USE_SEGGER_RTT in the menuconfig/guiconfig interface. The Application Development Primer, Setting
+Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful
+too.
+...
+error: Aborting due to Kconfig warnings
+
+CMake Error at /path/to/workspace/deps/zephyr/cmake/modules/kconfig.cmake:343 (message):
+  command failed with return code: 1
+Call Stack (most recent call first):
+  /path/to/workspace/deps/zephyr/cmake/modules/zephyr_default.cmake:115 (include)
+  /path/to/workspace/deps/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
+  /path/to/workspace/deps/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:97 (include_boilerplate)
+  CMakeLists.txt:2 (find_package)
+
+
+-- Configuring incomplete, errors occurred!
+FATAL ERROR: command exited with status 1: /opt/homebrew/bin/cmake -DWEST_PYTHON=/path/to/workspace/.venv/bin/python -B/path/to/workspace/build -GNinja -S/path/to/workspace/app
+```
+
+The build process was now able to correctly determine the `host-tools` and `toolchains`, but the build still failed with an error message that is quite hard to comprehend. Something seems to be missing!
+
+
+### Manifest imports
+
+The build fails since we're only adding the _Zephyr_ repository as a dependency. This is not enough: The Zephyr repository has its own dependencies, which it lists as _projects_ in its own `west.yml` file. We can _recursively import_ the required projects from Zephyr using the `import` key as follows:
+
+`app/west.yml`
+
+```yaml
+manifest:
+  version: 0.8
+
+  self:
+    path: app
+
+  projects:
+    - name: zephyr
+      revision: v3.4.0
+      url: https://github.com/zephyrproject-rtos/zephyr
+      # the path is no longer needed since we're now using `path-prefix`
+      # path: deps/zephyr
+      # add Zephyr-specific West extensions
+      west-commands: scripts/west-commands.yml
+      # recursively import Zephyr dependencies
+      import:
+        path-prefix: deps
+        file: west.yml
```

Comment:
Might be good to note that this is not strictly needed as this is the default value